test/src/Trait/TestFunctionalTrait.php

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+
+/*
+ Copyright (c) 2022, Manticore Software LTD (https://manticoresearch.com)
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License version 2 or any later
+ version. You should have received a copy of the GPL license along with this
+ program; if you did not, you can find it at http://www.gnu.org/
+ */
+
+namespace Manticoresearch\BuddyTest\Trait;
+
+use \Exception;
+
+trait TestFunctionalTrait {
+
+	/**
+	 * @var int $manticorePid
+	 */
+	protected static ?int $manticorePid = null;
+	/**
+	 * @var int $buddyPid
+	 */
+	protected static ?int $buddyPid = null;
+
+	/**
+	 * Launch daemon as as setup stage
+	 * @return void
+	 */
+	public static function setUpBeforeClass(): void {
+		$refCls = new \ReflectionClass(static::class);
+		$configFile = dirname((string)$refCls->getFileName()) . '/config/manticore.conf';
+		self::checkManticorePathes($configFile);
+		system("searchd --config '$configFile'");
+		self::$manticorePid = (int)trim((string)file_get_contents('/var/run/manticore-test/searchd.pid'));
+		sleep(2); // <- give 2 secs to protect from any kind of lags
+		self::loadBuddyPid();
+	}
+
+	/**
+	 *
+	 * @return void
+	 */
+	public static function tearDownAfterClass(): void {
+		system('pkill -9 searchd');
+		// To be sure run again kills for each pid
+		system('kill -9 ' . self::$manticorePid . ' 2> /dev/null');
+		system('kill -9 ' . self::$buddyPid . ' 2> /dev/null');
+	}
+
+	/**
+	 * Helper that checks if the pathes defined in the manticore config exist
+	 * and tries to create them if not
+	 *
+	 * @param string $configFile
+	 *
+	 * @return void
+	 */
+	protected static function checkManticorePathes($configFile): void {
+		$conf = file_get_contents($configFile);
+		if ($conf === false) {
+			throw new Exception("Unvalid Manticore config found at $configFile");
+		}
+		$propNames = ['log', 'query_log', 'pid_file', 'data_dir'];
+		foreach ($propNames as $prop) {
+			preg_match("/$prop = (.*?)(\/[^\/\r\n]*|)(\r|\n)/", $conf, $matches);
+			if (!isset($matches[1], $matches[2])) {
+				continue;
+			}
+			$checkDir = ($prop === 'data_dir') ? $matches[1] . $matches[2] : $matches[1];
+			if (is_dir($checkDir)) {
+				continue;
+			}
+			system("mkdir $checkDir 2>/dev/null", $res);
+			if ($res !== 0) {
+				throw new Exception("Cannot create Manticore `$prop` dir at $checkDir");
+			}
+		}
+	}
+
+	/**
+	 * Helper that allows us to reload fresh pid for relaunched
+	 * Buddy process by the Manticore
+	 * @return void
+	 */
+	protected static function loadBuddyPid(): void {
```

Comment:
This function duplicates the same one that was used in kill tests.

I have merged that branch; let's rebase and eliminate duplications of the same code.